harness/determined/keras/_tf_keras_trial.py

```diff
@@ -471,61 +488,92 @@ def _save_checkpoint(self, path: pathlib.Path) -> workload.Response:
+            rng_state["tf_rng_global_seed"] = tf.compat.v1.random.get_seed(0)[0]
         else:
             generator = tf.random.get_global_generator()
             rng_state["tf2_rng_global_algorithm"] = generator.algorithm
             rng_state["tf2_rng_global_state"] = generator.state
         with open(path.joinpath("rng_state.pkl"), "wb") as f:
             pickle.dump(rng_state, f)
 
+        # Save user code.
+        det.util.write_user_code(path)
+
+        # Save callback(s) state.
         callbacks_state = self.multiplexer._get_state()
         with path.joinpath("determined-callbacks.v1.pkl").open("wb") as f:
             pickle.dump(callbacks_state, f)
 
-        det.util.write_user_code(path)
-
         self.multiplexer._checkpoint_end(path)
 
-        return {"framework": f"tensorflow-{tf.__version__}", "format": "h5"}
+        return {"framework": f"tensorflow-{tf.__version__}", "format": "saved_weights"}
 
     def _load(self) -> None:
         self.multiplexer_load_state = None  # type: Optional[Dict]
         if not self.load_path:
             return
 
-        # Load model.
+        # Find model code path, we check multiple naming conventions for backwards compatibility.
         if self.load_path.joinpath("determined-keras-model.h5").exists():
-            full_ckpt_path = self.load_path.joinpath("determined-keras-model.h5")
+            model_weights_checkpoint_path = self.load_path.joinpath("determined-keras-model.h5")
+            optimizer_weights_checkpoint_path = model_weights_checkpoint_path
+        elif self.load_path.joinpath("determined-keras-optimizer-weights.h5").exists():
+            model_weights_checkpoint_path = self.load_path.joinpath(
+                "determined-keras-model-weights"
+            )
+            optimizer_weights_checkpoint_path = self.load_path.joinpath(
+                "determined-keras-optimizer-weights.h5"
+            )
         else:
-            full_ckpt_path = self.load_path.joinpath("determined-keras-model")
+            model_weights_checkpoint_path = self.load_path.joinpath("determined-keras-model")
+            optimizer_weights_checkpoint_path = model_weights_checkpoint_path
+
+        # Load model weights.
+        logging.info(f"Restoring model weights from {model_weights_checkpoint_path}.")
+        self.model.load_weights(str(model_weights_checkpoint_path))
 
-        logging.info(f"Restoring checkpoint from {full_ckpt_path}")
-        self.model.load_weights(str(full_ckpt_path))
-        load_optimizer_weights(self.model, full_ckpt_path)
+        # Load optimizer(s) weights.
+        logging.info(f"Restoring optimizer weights from {optimizer_weights_checkpoint_path}.")
+        with h5py.File(optimizer_weights_checkpoint_path, "r") as h5file:
+            if "optimizer_weights" in h5file:
+                load_optimizer_weights(self.model, h5file["optimizer_weights"], None)
+                return
```

Comment:
`return` seems like not the right thing to do here, what is wrong with an `else`?